sqrl-planner/src/main/java/com/datasqrl/plan/global/DAGAssembler.java

```diff
@@ -216,14 +215,16 @@ private RelNode produceWriteTree(RelNode relNode, SqrlConverterConfig config, Er
     AnnotatedLP alp = sqrlConverter.convert(relNode, config, errors);
     RelNode convertedRelNode = alp.getRelNode();
     //Expand to full tree
-    return produceWriteTree(convertedRelNode, alp.getTimestamp().getOnlyCandidate());
+    return produceWriteTree(convertedRelNode, alp.getTimestamp().getOnlyCandidateOptional());
   }
 
-  private RelNode produceWriteTree(RelNode convertedRelNode, int timestampIndex) {
+  private RelNode produceWriteTree(RelNode convertedRelNode, OptionalInt timestampIndex) {
     RelNode expandedRelNode = RelStageRunner.runStage(STREAM_DAG_STITCHING, convertedRelNode,
         framework.getQueryPlanner().getPlanner());
-    TimestampHint timestampHint = new TimestampHint(timestampIndex);
-    expandedRelNode = timestampHint.addHint((Hintable) expandedRelNode);
+    if (timestampIndex.isPresent()) {
```

Comment:
timestampIndex.ifPresent(...) is idomatic for optional

Reply:
need to reference expandedRelNode which cannot be made final.